### ISSUE TITLE
DeserializationError when using Panel 1.0.0rc7 with hvplot and widgets as arguments

### ISSUE THREAD

**Issue by MarcSkovMadsen** [holoviz/panel#4825]
panel==1.0.0rc7 hvplot==0.8.3 holoviews==1.16.0 bokeh==3.1.1

```python
import hvplot.pandas
import panel as pn

pn.extension()

from bokeh.sampledata.iris import flowers

x = pn.widgets.Select(name="x", options=["sepal_width", "petal_width"])
y = pn.widgets.Select(name="y", options=["sepal_length", "petal_length"])
kind = pn.widgets.Select(name="kind", value="scatter", options=["bivariate", "scatter"])

plot = flowers.hvplot(x=x, y=y, kind=kind, colorbar=False, width=600)
pn.Row(pn.WidgetBox(x, y, kind), plot).servable()
```

Serve the app and use a widget to change a value.

https://github.com/holoviz/panel/assets/42288570/00f63065-df78-4e56-87e4-1e2e66149524

```bash
$ panel serve script.py --autoreload
2023-05-12 07:19:19,759 W-1005 (FIXED_SIZING_MODE): 'fixed' sizing mode requires width and height to be set: Row(id='p1017', ...)
2023-05-12 07:19:19,794 Starting Bokeh server version 3.1.1 (running on Tornado 6.3.1)
2023-05-12 07:19:19,796 User authentication hooks NOT provided (default user enabled)
2023-05-12 07:19:19,798 Bokeh app running at: http://localhost:5006/script
2023-05-12 07:19:19,799 Starting Bokeh server with process id: 5940
2023-05-12 07:19:26,933 W-1005 (FIXED_SIZING_MODE): 'fixed' sizing mode requires width and height to be set: Row(id='p1204', ...)
2023-05-12 07:19:27,202 WebSocket connection opened
2023-05-12 07:19:27,204 ServerConnection created
2023-05-12 07:19:29,270 error handling message
 message: Message 'PATCH-DOC' content: {'events': [{'kind': 'ModelChanged', 'model': {'id': 'p1208'}, 'attr': 'inner_height', 'new': 0}, {'kind': 'ModelChanged', 'model': {'id': 'p1208'}, 'attr': 'outer_height', 'new': 55}]}
 error: DeserializationError("can't resolve reference 'p1208'")
Traceback (most recent call last):
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\server\protocol_handler.py", line 97, in handle
    work = await handler(message, connection)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\server\session.py", line 94, in _needs_document_lock_wrapper
    result = func(self, *args, **kwargs)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\server\session.py", line 288, in _handle_patch
    message.apply_to_document(self.document, self)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\protocol\messages\patch_doc.py", line 104, in apply_to_document
    invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.payload, setter=setter))
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\document\callbacks.py", line 443, in invoke_with_curdoc
    return f()
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\protocol\messages\patch_doc.py", line 104, in <lambda>
    invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.payload, setter=setter))
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\document\document.py", line 369, in apply_json_patch
    patch: PatchJson = deserializer.deserialize(patch_json)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 501, in deserialize
    return self.decode(obj.content, obj.buffers)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 516, in decode
    return self._decode(obj)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 557, in _decode
    return {key: self._decode(val) for key, val in obj.items()}
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 557, in <dictcomp>
    return {key: self._decode(val) for key, val in obj.items()}
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 559, in _decode
    return [self._decode(entry) for entry in obj]
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 559, in <listcomp>
    return [self._decode(entry) for entry in obj]
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 557, in _decode
    return {key: self._decode(val) for key, val in obj.items()}
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 557, in <dictcomp>
    return {key: self._decode(val) for key, val in obj.items()}
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 555, in _decode
    return self._decode_ref(cast(Ref, obj))
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 569, in _decode_ref
    self.error(f"can't resolve reference '{id}'")
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 717, in error
    raise DeserializationError(message)
bokeh.core.serialization.DeserializationError: can't resolve reference 'p1208'
2023-05-12 07:19:29,286 error handling message
 message: Message 'PATCH-DOC' content: {'events': [{'kind': 'ModelChanged', 'model': {'id': 'p1208'}, 'attr': 'inner_width', 'new': 0}, {'kind': 'ModelChanged', 'model': {'id': 'p1208'}, 'attr': 'outer_width', 'new': 73}]}
 error: DeserializationError("can't resolve reference 'p1208'")
Traceback (most recent call last):
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\server\protocol_handler.py", line 97, in handle
    work = await handler(message, connection)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\server\session.py", line 94, in _needs_document_lock_wrapper
    result = func(self, *args, **kwargs)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\server\session.py", line 288, in _handle_patch
    message.apply_to_document(self.document, self)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\protocol\messages\patch_doc.py", line 104, in apply_to_document
    invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.payload, setter=setter))
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\document\callbacks.py", line 443, in invoke_with_curdoc
    return f()
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\protocol\messages\patch_doc.py", line 104, in <lambda>
    invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.payload, setter=setter))
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\document\document.py", line 369, in apply_json_patch
    patch: PatchJson = deserializer.deserialize(patch_json)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 501, in deserialize
    return self.decode(obj.content, obj.buffers)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 516, in decode
    return self._decode(obj)
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 557, in _decode
    return {key: self._decode(val) for key, val in obj.items()}
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 557, in <dictcomp>
    return {key: self._decode(val) for key, val in obj.items()}
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 559, in _decode
    return [self._decode(entry) for entry in obj]
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 559, in <listcomp>
    return [self._decode(entry) for entry in obj]
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 557, in _decode
    return {key: self._decode(val) for key, val in obj.items()}
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 557, in <dictcomp>
    return {key: self._decode(val) for key, val in obj.items()}
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 555, in _decode
    return self._decode_ref(cast(Ref, obj))
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 569, in _decode_ref
    self.error(f"can't resolve reference '{id}'")
  File "C:\repos\private\tmp\.venv\lib\site-packages\bokeh\core\serialization.py", line 717, in error
    raise DeserializationError(message)
bokeh.core.serialization.DeserializationError: can't resolve reference 'p1208'
```

**Comment by philippjfr** [comment]
@mattpap can we just ignore these on the bokeh side please? I know we do so on JS but it does seem like we never did in Python.

**Comment by mattpap** [comment]
> can we just ignore these on the bokeh side please? I know we do so on JS but it does seem like we never did in Python.

`can't resolve reference` is a different kind of error than `reference is already known`. The later can be ignored, because there's an instance to use, but the former has nothing to work with (it can't just return `None` or something like this).

Looking at the error message:

> bokeh.core.serialization.DeserializationError: can't resolve reference 'p1208'

and the message contents:

> `{'events': [{'kind': 'ModelChanged', 'model': {'id': 'p1208'}, 'attr': 'inner_height', 'new': 0}, {'kind': 'ModelChanged', 'model': {'id': 'p1208'}, 'attr': 'outer_height', 'new': 55}]}`

we can see that the offending reference is the model that the message is targetting. This perhaps could be ignored, but I'm pretty sure this is an indicator of a bug, especially given how simple the example is and the contents of the message (there should be four properties modified: `inner_{width,height}` and `outer_{width,height}`; also the values look strange). That's assuming that `hvplot` doesn't do some magic behind this otherwise straightforward API.

**Comment by philippjfr** [comment]
I think all that's happening here is that the figure is replaced in it's entirety and then messages from the old figure arrive.

**Comment by mattpap** [comment]
When changing a value of a select widget, the following sequence of events happen:
```
bokeh.js:1093 Row(p3369).sizing_mode = null
bokeh.js:43866 after_layout() Figure(p3373)
bokeh.js:1093 Figure(p3373).height = null
bokeh.js:43866 after_layout() Figure(p3373)
bokeh.js:1093 Figure(p3373).width = null
bokeh.js:43866 after_layout() Figure(p3373)
bokeh.js:1093 Figure(p3373).sizing_mode = null
bokeh.js:43866 after_layout() Figure(p3373)
bokeh.js:1093 Row(p3369).children = Figure(p3593)
bokeh.js:43866 after_layout() Figure(p3593)
```
First the existing plot is being changed (four times and in a peculiar way), then the last change triggers layout and `after_layout()`, which then updates `inner_*` and `outer_*` (with peculiar values due to `null`s). These messages arrive when the plot being changed is already removed.

I'm guessing that hvplot or holoviews is responsible for those `null`s. Why does it do this? In any case, we don't have tests in bokeh for the situation where a model is changed (and triggers layout) just before being removed, thus it wasn't caught earlier.

**Comment by MarcSkovMadsen** [comment]
I see this issue also when using hvplot in other ways

```bash
2023-05-12 14:42:36,757 error handling message
 message: Message 'PATCH-DOC' content: {'events': [{'kind': 'ModelChanged', 'model': {'id': 'p1490'}, 'attr': 'inner_height', 'new': 360}]} 
 error: DeserializationError("can't resolve reference 'p1490'")
Traceback (most recent call last):
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/server/protocol_handler.py", line 97, in handle
    work = await handler(message, connection)
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/server/session.py", line 94, in _needs_document_lock_wrapper
    result = func(self, *args, **kwargs)
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/server/session.py", line 288, in _handle_patch
    message.apply_to_document(self.document, self)
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/protocol/messages/patch_doc.py", line 104, in apply_to_document
    invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.payload, setter=setter))
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/document/callbacks.py", line 443, in invoke_with_curdoc
    return f()
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/protocol/messages/patch_doc.py", line 104, in <lambda>
    invoke_with_curdoc(doc, lambda: doc.apply_json_patch(self.payload, setter=setter))
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/document/document.py", line 369, in apply_json_patch
    patch: PatchJson = deserializer.deserialize(patch_json)
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/core/serialization.py", line 501, in deserialize
    return self.decode(obj.content, obj.buffers)
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/core/serialization.py", line 516, in decode
    return self._decode(obj)
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/core/serialization.py", line 557, in _decode
    return {key: self._decode(val) for key, val in obj.items()}
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/core/serialization.py", line 557, in <dictcomp>
    return {key: self._decode(val) for key, val in obj.items()}
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/core/serialization.py", line 559, in _decode
    return [self._decode(entry) for entry in obj]
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/core/serialization.py", line 559, in <listcomp>
    return [self._decode(entry) for entry in obj]
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/core/serialization.py", line 557, in _decode
    return {key: self._decode(val) for key, val in obj.items()}
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/core/serialization.py", line 557, in <dictcomp>
    return {key: self._decode(val) for key, val in obj.items()}
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/core/serialization.py", line 555, in _decode
    return self._decode_ref(cast(Ref, obj))
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/core/serialization.py", line 569, in _decode_ref
    self.error(f"can't resolve reference '{id}'")
  File "/home/jovyan/repos/mt-gas-fundamentals/.venv/lib/python3.10/site-packages/bokeh/core/serialization.py", line 717, in error
    raise DeserializationError(message)
bokeh.core.serialization.DeserializationError: can't resolve reference 'p1490'
```

**Comment by philippjfr** [comment]
Yes to be clear I see the deserialization issue all the time in all kinds of scenarios. It doesn't generally cause issues but still, I think receiving a message about a model that was already removed can safely be removed.

**Comment by philippjfr** [comment]
>When changing a value of a select widget, the following sequence of events happen:

```
bokeh.js:1093 Row(p3369).sizing_mode = null
bokeh.js:43866 after_layout() Figure(p3373)
bokeh.js:1093 Figure(p3373).height = null
bokeh.js:43866 after_layout() Figure(p3373)
bokeh.js:1093 Figure(p3373).width = null
bokeh.js:43866 after_layout() Figure(p3373)
bokeh.js:1093 Figure(p3373).sizing_mode = null
bokeh.js:43866 after_layout() Figure(p3373)
bokeh.js:1093 Row(p3369).children = Figure(p3593)
bokeh.js:43866 after_layout() Figure(p3593)
```

I actually think the `width=null` events are originating on the frontend because I do not see the backend sending these changes.

**Comment by philippjfr** [comment]
Okay, this was (partially) Panel's fault, basically we were dispatching a bunch of events on a model that was already being swapped out, causing the deserialization errors (fixed in #4846). I still think Bokeh should handle this to avoid such errors.

However the actual reported issue here still remains (confirming my suspicion that the deserialization errors were unrelated). Basically it seems like while the figure is being swapped out Bokeh computes an `inner_height` of zero resulting in the plot to collapse.

**Comment by philippjfr** [comment]
Nevermind, sorry for all the noise here. The height collapsing was also caused by Panel.
